tests/test_diffraction_objects.py

```diff
@@ -369,6 +371,29 @@ def test_all_array_setter():
         actual_do.all_arrays = np.empty((4, 4))
 
 
+def test_id_getter():
+    do = DiffractionObject()
+    assert hasattr(do, "id")
+    assert isinstance(do.id, UUID)
+    assert len(str(do.id)) == 36
+
+
+def test_id_getter_with_mock(mocker):
+    mocker.patch.object(DiffractionObject, "id", new_callable=lambda: UUID("d67b19c6-3016-439f-81f7-cf20a04bee87"))
```

Comment:
Since `id` is used with `@property`, the following doesn't apply:

```
mocker.patch.object(DiffractionObject, 'id', return_value="d67b19c6-3016-439f-81f7-cf20a04bee87")
  ```

Hence we use the `new_callable` parameter.

Reply:
I see, nicely done.  I didn't know about that.